src/core/Akka.Streams/Dsl/Internal/InternalFlowOperations.cs

```diff
@@ -1999,7 +1999,7 @@ public static IFlow<(T1, long), TMat> ZipWithIndex<T1, TMat>(this IFlow<T1, TMat
         /// <param name="segmentSize">TBD</param>
         /// <returns>TBD</returns>
         public static IFlow<T2, TMat> Interleave<T1, T2, TMat>(this IFlow<T1, TMat> flow,
-            IGraph<SourceShape<T2>, TMat> graph, int segmentSize) where T1 : T2
+            IGraph<SourceShape<T2>, TMat> graph, int segmentSize) where T1 : T1
```

Comment:
I think "T1 : T1" should be "T2 : T1"

:)

Reply:
Yeah, I thought so too, but this line will not compile

https://github.com/akkadotnet/akka.net/blob/0948525ff24f68fa939142f9d41f59744c0cd2ff/src/core/Akka.Streams/Dsl/Internal/InternalFlowOperations.cs#L2045

seems like we somehow use this "base to child" connection. Probably we are missing some core idea here...

Reply:
Does it have any value leaving the constraint?

Ie. will it not compile or cause other issues if the constraint is simply removed?

Reply:
Will not compile. Feel free to continue my work here by the way - could not change this code quickly enough to have this inheritance constraints swapped... Have a feeling that it works here due to some hack in graph architecture, but could not catch it.

Reply:
Thank you! I will give it a go!

Reply:
I saw this in the build logs: 

https://dev.azure.com/dotnet/Akka.NET/_build/results?buildId=35831&view=logs&j=d83928d8-8bfe-51de-721e-9025f72ee732&t=21d77f07-2bef-56ac-91e9-8618e6456479&l=1178

I just want to confirm if this is fine and if we need some compiler flag for T1 : T1 to be accepted. I got the same in my local build. Otherwise, I can try removing the constraint and see what I manage.

Let me know!